### ISSUE TITLE
Check to see if services are still running

### ISSUE THREAD

**Issue by elimisteve** [elimisteve/hyperactive#2]
**NOTE: I'm not worrying about making this site checker federated/p2p until later.**

Protocols to support:
- [ ] HTTP/S
- [ ] SMTP (requested by @thefinn93)
- [ ] IMAP (requested by @thefinn93)
- [ ] SIP (requested by @thefinn93)

**Comment by thefinn93** [comment]
- [ ] Jabber/XMPP

**Comment by thefinn93** [comment]
- [ ] Outdated CJDNS version (for people like @elimisteve)

**Comment by thefinn93** [comment]
Maybe make it just use nagios plugins, then you won't need to worry about supporting every random-ass service

**Comment by elimisteve** [comment]
@thefinn93 Yeah that's true... something lightweight meant for federation may make sense, though it looks like Nagios can be federated, too.  Hmm.

**Comment by thefinn93** [comment]
Oh i didn't mean full nagios. Nagios checks a service by executing a command and parsing the output (not much parsing, but yeah). I'm saying, make Hyperactive compatible with these plugins.

**Comment by thefinn93** [comment]
Actually it's primarily based on exit status, so if the check command exits with status 0 the thing is okay, 1 the thing is WARNING, and 2 it's ERROR. Anything else is plugin error. The first line of stdout is the status line to give more description of what's wrong

**Comment by thefinn93** [comment]
See [Nagios Exchange](https://exchange.nagios.org/directory/Plugins) for a whole shitload of these things

**Comment by elimisteve** [comment]
- [ ] Check Hypemail, IRC, and other major sites/services (idea from elwisp)

**Comment by kylerchin** [comment]
How is this useful when your service is down?
